src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -186,22 +196,108 @@ public void close() throws IOException
       {
         open = false;
         internalFiles.put(name, new Metadata(currOut.getFileNum(), startOffset, currOut.getCurrOffset()));
+        writerCurrentlyInUse = false;
 
         if (bytesWritten != currOut.getCurrOffset() - startOffset) {
           throw new ISE("WTF? Perhaps there is some concurrent modification going on?");
         }
         if (bytesWritten != size) {
           throw new IOException(
               String.format("Expected [%,d] bytes, only saw [%,d], potential corruption?", size, bytesWritten)
-          );
+              );
+        }
+
+        if(!writerCurrentlyInUse)
```

Comment:
Given that you just set `writerCurrentlyInUse` to `false` and that this object is not attempting to be thread-safe, I don't think this can ever be false.